src/Resolvers/Microsoft.DotNet.NativeWrapper/EnvironmentProvider.cs

```diff
@@ -89,5 +89,51 @@ public static string GetDotnetExeDirectory(Func<string, string> getEnvironmentVa
             var environmentProvider = new EnvironmentProvider(getEnvironmentVariable);
             return environmentProvider.GetDotnetExeDirectory();
         }
+
+        public string GetUserHomeDirectory()
+        {
+            var home = _getEnvironmentVariable("DOTNET_CLI_HOME");
+            if (string.IsNullOrEmpty(home))
+            {
+                home = _getEnvironmentVariable(Interop.RunningOnWindows ? "USERPROFILE" : "HOME");
+                if (string.IsNullOrEmpty(home))
+                {
+                    throw new Exception("Unable to resolve user home");
+                }
+            }
+
+            return home;
+        }
+
+        public static string GetUserHomeDirectory(Func<string, string> getEnvironmentVariable = null)
+        {
+            if (getEnvironmentVariable == null)
+            {
+                getEnvironmentVariable = Environment.GetEnvironmentVariable;
+            }
+            var environmentProvider = new EnvironmentProvider(getEnvironmentVariable);
+            return environmentProvider.GetUserHomeDirectory();
+        }
+
+        public string GetTempDirectory()
+        {
+            var tmp = _getEnvironmentVariable(Interop.RunningOnWindows ? "TEMP" : "TMPDIR");
+            if (string.IsNullOrEmpty(tmp))
+            {
+                throw new Exception("Unable to resolve user temp");
```

Comment:
Should these messages be localized or do users never see them?

Reply:
I think we generally don't localize error messages since theoretically the user won't ever see them (if everything goes right :) )

Reply:
Hmm, not buying that completely. If users don't see errors, then there won't be a point in reporting a message :) It can impact accessibility, e.g. screen readers that assist non-English speaking users.